lib/urldata.h

```diff
@@ -855,6 +855,7 @@ struct Curl_handler {
 #define PROTOPT_STREAM (1<<9) /* a protocol with individual logical streams */
 #define PROTOPT_URLOPTIONS (1<<10) /* allow options part in the userinfo field
                                       of the URL */
+#define PROTOPT_HTTP_PROXY (1<<11) /* allow over a HTTP proxy */
```

Comment:
I think we should explain that this flag is not intended for HTTP itself, and that the only difference is whether tunneling will be enabled automatically.

Maybe "HTTP proxies may know this protocol and act as a gateway" ?